miden-lib/src/notes/mod.rs

```diff
@@ -50,6 +57,27 @@ pub fn create_note(
             ProgramAst::from_bytes(p2idr_bytes).map_err(NoteError::NoteDeserializationError)?,
             vec![target.into(), recall_height.into(), ZERO, ZERO],
         ),
+        Script::ASWAP {
+            requested_asset,
+            tag,
+            recipient,
+        } => (
+            ProgramAst::from_bytes(aswap_bytes).map_err(NoteError::NoteDeserializationError)?,
+            vec![
```

Comment:
Here we now should put a function that creates this input vector for us. the function should take `serial_num` and `ASSET` and return 

`vec![RECIPIENT, tag, ASSET]`

whereas `RECIPIENT` is computed as `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)`: 

- `script_hash` should be the hash of the `P2ID` script
- `input_hash` the hash of the inputs to the `P2ID` script (which is the `target_account_id` (in your case the account ID of the SENDER of the ASWAP script - which is the target of the return note)

- `tag` can be received from the Account-ID from the sender of the ASWAP script.

Reply:
I believe that this is how i would create my recipient on the rust side:

https://github.com/0xPolygonMiden/miden-base/blob/9806a922a3b99948f6254f8ae105904dd4eda7ea/objects/src/notes/mod.rs#L149-L155C9

Reply:
exactly